pkg/config/types.go

```diff
@@ -0,0 +1,79 @@
+package config
+
+import (
+	"errors"
+	"time"
+
+	"github.com/Azure/aks-app-routing-operator/pkg/util"
+)
+
+// ControllerConfig specifies configuration options for an Ingress Controller
+type ControllerConfig int
+
+const (
+	// Standard means no special configuration rules are applied to the Controller
+	Standard ControllerConfig = iota
+	// Public means the Ingress Controller is exposed externally and is public facing
+	Public
+	// Private means the Ingress Controller is internally facing and is backed by a private IP address
+	Private
+	// Off means the Ingress Controller isn't used
+	Off
+)
+
+var controllerConfigMapping = map[ControllerConfig]string{
+	Standard: "standard",
+	Public:   "public",
+	Private:  "private",
+	Off:      "off",
+}
+
+func (c *ControllerConfig) String() string {
+	if c == nil {
+		return "nil"
+	}
+
+	if str, ok := controllerConfigMapping[*c]; ok {
+		return str
+	}
+
+	return "unknown"
```

Comment:
do we want to return standard here instead of a new "unknown" type if we can't recognize the enum? that seems like the safer path

Reply:
this is just the stringify method. returning an unknown string is appropriate. we properly return an error in the parsing logic